modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/LeaseUpdater.java

```diff
@@ -412,7 +412,7 @@ private void updateLeaseBatchInternal() {
                 if (!lease.isAccepted()) {
                     LeaseAgreement agreement = leaseNegotiator.getAndRemoveIfReady(grpId);
 
-                    agreement.checkValid(grpId, topologyTracker.currentTopologySnapshot(), assignments);
+                    agreement.checkValid(grpId, topologyTracker.currentTopologySnapshot(), stableAssignments);
```

Comment:
Hm, I'm not sure there: lease on pendings set is an exceptional situation and I guess It's a good idea to invalidate pending lease and grant it to a node from stables. What do you think?

Reply:
I mean the folowing:
- no stable assignments left, the candidate is chosen from pendings
- lease agreement is created and negotiation starts
- on the next iteration `agreement.checkValid` may see that the candidate is not in the given assignments anymore (and the given assignments are stable here), and the candidate  never was in the stable assignments because if was chosen from pendings. However, `checkValid` knows nothing about it and invalidates the agreement, stopping the negotiation and beginning the lease granting process from the start. This may cause flakiness of tests.